check_packages.py

```diff
@@ -1,99 +1,124 @@
-    'lxml': {'version': '4.3', 'import_name': 'lxml.etree', 'test': 'LXML_VERSION'},
-    'PIL': {'version': '1.1.7', 'import_name': 'Image', 'test': 'VERSION'},
-    'pylint': {'version': None, 'import_name': 'pylint', 'test': None},
-    'periodictable': {'version': '1.5.0', 'import_name': 'periodictable', 'test': '__version__'},
-    'bumps': {'version': '0.7.11', 'import_name': 'bumps', 'test': '__version__'},
-    'numpy': {'version': '1.7.1', 'import_name': 'numpy', 'test': '__version__'},
-    'scipy': {'version': '1.2.0', 'import_name': 'scipy', 'test': '__version__'},
-    'pyqt': {'version': '5.6', 'import_name': 'PyQt5.QtCore', 'test': 'PYQT_VERSION_STR'},
-    'matplotlib': {'version': '2.2.0', 'import_name': 'matplotlib', 'test': '__version__'},
-    'xhtml2pdf': {'version': '3.0.33', 'import_name': 'xhtml2pdf', 'test': '__version__'},
-    'sphinx': {'version': '1.2.1', 'import_name': 'sphinx', 'test': '__version__'},
-    'pyopencl': {'version': '2018.2', 'import_name': 'pyopencl', 'test': 'VERSION_TEXT'},
-    'ipykernel': {'version': '4.9.0', 'import_name': 'ipykernel', 'test': '__version__'},
-    'ipython': {'version': '6.5.0', 'import_name': 'IPython', 'test': '__version__'},
-    'jupyter': {'version': '1.0.0', 'import_name': 'jupyter', 'test': '__version__'},
-    'twisted': {'version': '18.9.0', 'import_name': 'twisted', 'test': '__version__'},
-    'pytest': {'version': '4.2.0', 'import_name': 'pytest', 'test': '__version__'},
-}
-win_required_package_list = {
-}
+# Output strings
+CORRECT = Style.RESET_ALL + " - {0} - Expected Version Installed: {1}"
+VERSION_MISMATCH = Fore.LIGHTYELLOW_EX +\
+          " - {0} - Version Mismatch - Installed: {1}, Expected: {2}"
+VERSION_UNKNOWN = Fore.YELLOW +\
+                  " - {0} - Version Cannot Be Determined - Expected: {1}"
+NOT_INSTALLED = Fore.LIGHTRED_EX + " - {0} NOT INSTALLED"
 
-deprecated_package_list = {
-}
-
-print("Checking Required Package Versions....\n")
-print("Common Packages")
+# Location of yaml files
+if sys.platform == 'win32':
+    file_location = './build_tools/conda_qt5_win.yml'
+elif sys.platform == 'darwin':
+    file_location = './build_tools/conda_qt5_osx.yml'
+else:
+    file_location = './build_tools/conda_qt5_ubuntu.yml'
 
-for package_name, test_vals in common_required_package_list.items():
+# Open yaml and get packages
+with open(file_location, 'r') as stream:
     try:
-        i = __import__(test_vals['import_name'], fromlist=[''])
-        if test_vals['test'] is None:
-            print("%s Installed (Unknown version)" % package_name)
-        elif package_name == 'lxml':
-            verstring = str(getattr(i, 'LXML_VERSION'))
-            print("%s Version Installed: %s"% (package_name, verstring.replace(', ', '.').lstrip('(').rstrip(')')))
+        yaml_dict = yaml.load(stream, Loader=yaml.SafeLoader)
+        yaml_packages = yaml_dict['dependencies']
+        common_required_package_list = []
+        pip_packages = yaml_packages.pop()
```

Comment:
Trusting that 'pip' is the last entry in the list is a bit risky.  yaml will probably allow additional packages to be included after the pip package, but the process for generating the yaml file probably won't.

Maybe:
```python
inter_package_list = [p for entry in yaml_packages for p in (entry['pip'] if isinstance(entry, dict) else [entry])]
common_required_package_list = [re.sub("[<>=]", "=", package) for package in inter_package_list]  
```
or using for-loops:
```python
inter_package_list = []
for package in yaml_packages:
    if isinstance(package, dict):
        assert "pip" in package, f"yaml dependencies can only handle pip sub-lists not {package}"
        inter_package_list.extend(package['pip'])
    else:
        inter_package_list.append(package)
common_required_package_list = [re.sub("[<>=]", "=", package) for package in inter_package_list]  
```